src/core/abr/representation_estimator.ts

```diff
@@ -191,6 +194,47 @@ export default function RepresentationEstimator({
                                                                        initialBitrate,
                                               lowLatencyMode);
   const requestsStore = new PendingRequestsStore();
+  const latestDurationFromCache: { video: number|undefined;
+                                   audio: number|undefined; } = { video: undefined,
+                                                                  audio: undefined };
+
+  let hasLoadedNonCachedSegments = false;
+
+  /**
+   * Determines with request duration if a loaded chunk may have been loaded
+   * from cache
+   * @param {Object} content
+   * @param {number} duration
+   * @returns {undefined|boolean}
+   */
+  function mayBeFromCache(content: { representation: Representation;
+                                     adaptation: Adaptation;
+                                     segment: ISegment; },
+                          duration: number): undefined|boolean {
+    const contentType = content.adaptation.type;
+    if (contentType === "text" || contentType === "image") {
+      return undefined;
+    }
+
+    const [ isFromCacheMaxDuration,
+            canBeFromCacheMaxDuration ] =
+      CACHE_LOAD_DURATION_THRESHOLDS[contentType];
+
+    const latestDurationFromCacheByType =
+      latestDurationFromCache[contentType];
+    if (duration > canBeFromCacheMaxDuration ||
+        (
+          duration > isFromCacheMaxDuration &&
+          latestDurationFromCacheByType !== undefined &&
+          latestDurationFromCacheByType > isFromCacheMaxDuration
+        )
+    ) {
+      latestDurationFromCache[contentType] = undefined;
```

Comment:
From the name of the function, I wouldn't have guesses that it was it which mutated `latestDurationFromCache`.

I would have found it more readable to move that function elsewhere adds a `previousDownloadDuration : number | undefined` argument or something and then mutating the property here.

What do you think?